giskard_vision/object_detection/dataloaders/loaders.py

```diff
@@ -187,6 +195,251 @@ def get_image(self, idx: int) -> np.ndarray:
+                categories=[
+                    "gender",
+                    "glasses",
+                    "exposure_exposureLevel",
+                    "noise_noiseLevel",
+                    "makeup_eyeMakeup",
+                    "makeup_lipMakeup",
+                    "occlusion_foreheadOccluded",
+                    "occlusion_eyeOccluded",
+                    "occlusion_mouthOccluded",
+                    "hair_invisible",
+                    "hairColor",
+                    "emotion",
+                ],
+                issue_groups={
+                    "smile": PerformanceIssueMeta,
+                    "headPose_pitch": PerformanceIssueMeta,
+                    "headPose_roll": PerformanceIssueMeta,
+                    "headPose_yaw": PerformanceIssueMeta,
+                    "gender": EthicalIssueMeta,
+                    "age": EthicalIssueMeta,
+                    "facialHair_moustache": EthicalIssueMeta,
+                    "facialHair_beard": EthicalIssueMeta,
+                    "facialHair_sideburns": EthicalIssueMeta,
+                    "glasses": EthicalIssueMeta,
+                    "emotion": PerformanceIssueMeta,
+                    "blur_blurLevel": PerformanceIssueMeta,
+                    "blur_value": PerformanceIssueMeta,
+                    "exposure_exposureLevel": PerformanceIssueMeta,
+                    "exposure_value": PerformanceIssueMeta,
+                    "noise_noiseLevel": PerformanceIssueMeta,
+                    "noise_value": PerformanceIssueMeta,
+                    "makeup_eyeMakeup": EthicalIssueMeta,
+                    "makeup_lipMakeup": EthicalIssueMeta,
+                    "occlusion_foreheadOccluded": PerformanceIssueMeta,
+                    "occlusion_eyeOccluded": PerformanceIssueMeta,
+                    "occlusion_mouthOccluded": PerformanceIssueMeta,
+                    "hair_bald": EthicalIssueMeta,
+                    "hair_invisible": PerformanceIssueMeta,
+                    "hairColor": EthicalIssueMeta,
+                },
+            )
+        except FileNotFoundError:
+            return None
+
+
+class DataLoaderFFHQFaceDetectionLandmark(DataLoaderFFHQ):
+    """Data loader for the FFHQ (Flickr-Faces-HQ) dataset for face detection."""
+
+    def get_labels(self, idx: int) -> Optional[np.ndarray]:
+        """
+        Gets marks for a specific index after validation.
+        Args:
+            idx (int): Index of the data.
+        Returns:
+            Optional[np.ndarray]: Marks for the given index.
+        """
+        landmarks = super().get_labels(idx)
+
+        if landmarks is None:
+            return None
+
+        min_point = np.min(landmarks, axis=0)
+        max_point = np.max(landmarks, axis=0)
+
+        return {
+            "boxes": np.array([min_point[0], min_point[1], max_point[0], max_point[1]]),
+            "labels": "face",
+        }
+
+    def get_meta(self, idx: int) -> Optional[Dict[str, Any]]:
```

Comment:
I would avoid duplicating this here, can you base `DataLoaderFFHQFaceDetectionLandmark` on `DataLoaderFFHQFaceDetection`?

Reply:
Done. I did it in the contrary.

`DataLoaderFFHQFaceDetectionLandmark` uses the landmarks to create bboxes and keeps the meta, while `DataLoaderFFHQFaceDetection` rewrites both of them.